src/resolve/resolved-dnssd.c

```diff
@@ -37,7 +37,7 @@ DnssdTxtData *dnssd_txtdata_free_all(DnssdTxtData *txt_data) {
         return dnssd_txtdata_free_all(next);
 }
 
-DnssdService *dnssd_service_free(DnssdService *service) {
+DnssdRegisteredService *dnssd_service_free(DnssdRegisteredService *service) {
```

Comment:
Then, maybe it is better to rename dnssd_service_free() -> dnssd_registered_service_free() ?